<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Tutorial</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(PANE)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;dplyr&#39;</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#&gt;     filter, lag</span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(nnls)</span></code></pre></div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p><em>PANE</em> is an R package that leverages PCA and NNLS
(Non-Negative Least Squares) to assess the ancestral composition of
admixed individuals with high accuracy and reliability, the approach
further described in <a href>here</a>. PANE assigns the genetic
ancestry of list of target samples or populations given a list of
putative sources.</p>
<p><em>PANE</em> estimations are based on a PCA where the targets and
source groups are available, along with those we suggest to use
additional groups to better define the PC space. Once that the PC space
is defined, a set of NNLS is then applied on the PC coordinates,
effectively summarizing the genetic ancestry.</p>
<p>For its most basic usage PANE needs:</p>
<ol style="list-style-type: upper-roman">
<li>A PCA matrix (a dataframe with N PCs)</li>
<li>A list of target and source groups (or samples)</li>
</ol>
</div>
<div id="basic-usage-example" class="section level1">
<h1>Basic usage example</h1>
<p>Right below you can find a very basic way to run PANE, if you want
more details, please keep on reading.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="sc">$</span> pca <span class="ot">=</span> <span class="fu">read_eigen</span>(<span class="at">pca_input =</span> <span class="st">&#39;data/TOY.pca.evec&#39;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">=</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&#39;GST1&#39;</span>,<span class="st">&#39;GST2&#39;</span>), <span class="at">admixed=</span><span class="fu">c</span>(<span class="st">&#39;70GST1.30GST2&#39;</span>))</span></code></pre></div>
<p>PANE package has two functions to read the PCA matrix,
<code>read_eigen()</code> and <code>read_flash()</code>.<br />
* <code>read_eigen()</code> will read a PCA that has been created with
smartpca from the EIGENSOFT software.<br />
* <code>read_flash()</code> instead, will read a PCA that has been
created with flashpca software.<br />
The goal of both functions is to set the PCA file so that it looks as
follows:</p>
<table>
<thead>
<tr class="header">
<th align="left">POP</th>
<th align="center">IND</th>
<th align="center">PCN</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Source1</td>
<td align="center">IND_1</td>
<td align="center">0.01</td>
</tr>
<tr class="even">
<td align="left">Source1</td>
<td align="center">IND_2</td>
<td align="center">0.02</td>
</tr>
<tr class="odd">
<td align="left">Source2</td>
<td align="center">IND_1</td>
<td align="center">-0.03</td>
</tr>
<tr class="even">
<td align="left">Admix1</td>
<td align="center">IND_1</td>
<td align="center">0.08</td>
</tr>
</tbody>
</table>
<p>If neither <code>read_eigen()</code> nor <code>read_flash()</code> is
for you, you might want to simply use <code>read.table()</code>, and set
the file so that it has the aforementioned look.</p>
<p>For the sake of the example, let’s say you obtained a PCA from the
software EIGENSOFT, using smartpca.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="sc">$</span> pca <span class="ot">=</span> <span class="fu">read_eigen</span>(<span class="at">pca_input =</span> <span class="st">&#39;data/TOY.pca.evec&#39;</span>)</span></code></pre></div>
<p><em>NB</em>: The script assumes that there are family names available
in the first column of the PCA in this format: <a href="POP:ID" class="uri">POP:ID</a> (or FamilyName:SampleName)</p>
<p>The second step is to set up an ‘AS_file’, a file with the list of
the Admixed groups (A) and the Source groups (S). If you want to use
PANE sample-wise rather than group-wise, simply adjust the PCA file so
that in the ‘POP’ column is identical to the ‘IND’ column, and set the
AS_file with the samples list, rather than the group list.<br />
The AS_file is a two-columns file with the population list on the first
column, and the ‘A/S’ information on the second column. The ‘A/S’
information stands for Admixed (A) or Source (S), for each
population/group we will indicate whether PANE should consider it as a
Source (S) or as an admixed target (A), the file looks like this:</p>
<table>
<thead>
<tr class="header">
<th>POP A/S</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Source1</td>
<td>S</td>
</tr>
<tr class="even">
<td>Source2</td>
<td>S</td>
</tr>
<tr class="odd">
<td>Admix1</td>
<td>A</td>
</tr>
</tbody>
</table>
<p><em>NB</em>: If the column A/S does not contain “S” or “A”, the
population/ID will be skipped.</p>
<p>To read the AS_file, a simple <code>read_table(file, header=T)</code>
will be sufficient.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="sc">$</span> AS_file <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&#39;data/Example_AS&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>With the PCA and AS_file loaded, we are finally ready to run PANE as
follows:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">=</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">as_file =</span> AS_file)</span></code></pre></div>
<p>You can avoid relying on the AS_file if you wish, and rather use a
vector of the target and source groups directly in pane() function, as
follows:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">=</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&#39;GST1&#39;</span>,<span class="st">&#39;GST2&#39;</span>), <span class="at">admixed=</span><span class="fu">c</span>(<span class="st">&#39;70GST1.30GST2&#39;</span>))</span></code></pre></div>
<p>Finally, if you want to save <em>PANE</em> results on a table-like
format, you can use <code>write_pane()</code>, this way:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">&lt;-</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&#39;GST1&#39;</span>,<span class="st">&#39;GST2&#39;</span>), <span class="at">admixed=</span><span class="fu">c</span>(<span class="st">&#39;70GST1.30GST2&#39;</span>))</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="sc">$</span> <span class="fu">write_pane</span>(PANE_result, <span class="at">output_name =</span> <span class="st">&#39;my_dir/my_PANE_results.txt&#39;</span>)</span></code></pre></div>
</div>
<div id="plotting" class="section level1">
<h1>Plotting</h1>
<p>The function <code>plot_pane()</code> can help you plot
<code>pane()</code> results. Based on your preferences, you can either
rely on basic R or on ggplot2 to plot your results. Let’s say we want to
plot PANE results using ggplot2:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">&lt;-</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&#39;GST1&#39;</span>,<span class="st">&#39;GST2&#39;</span>), <span class="at">admixed=</span><span class="fu">c</span>(<span class="st">&#39;70GST1.30GST2&#39;</span>))</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="sc">$</span> <span class="fu">plot_pane</span>(PANE_result, <span class="st">&#39;PANE_plot&#39;</span>, <span class="at">type_ggplot =</span> <span class="st">&#39;YES&#39;</span>)</span></code></pre></div>
<p>Alternatively, to plot with basic R, just run the command without
<code>type_ggplot</code> option, like so:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="sc">$</span> PANE_result <span class="ot">&lt;-</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&#39;GST1&#39;</span>,<span class="st">&#39;GST2&#39;</span>), <span class="at">admixed=</span><span class="fu">c</span>(<span class="st">&#39;70GST1.30GST2&#39;</span>))</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="sc">$</span> <span class="fu">plot_pane</span>(PANE_result, <span class="st">&#39;PANE_plot&#39;</span>)</span></code></pre></div>
<p>You will find a barplot in your working directory.</p>
</div>
<div id="euclidean-distances-between-sources" class="section level1">
<h1>Euclidean Distances between Sources</h1>
<p>The function <code>pcs_distances</code> allows you to estimate and
visualize the Euclidean Distances (ED) between pairs of sources. To run
the function you need to provide:</p>
<ul>
<li><code>pca_input</code> takes a pca matrix</li>
<li><code>output_name</code> string with path an output file name</li>
<li><code>sources_file</code> two-columns table with the list of source
pairs to test:</li>
</ul>
<table>
<thead>
<tr class="header">
<th>S1 S1</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GST1</td>
<td>GST2</td>
</tr>
<tr class="even">
<td>GST2</td>
<td>GST3</td>
</tr>
<tr class="odd">
<td>GST3</td>
<td>GST4</td>
</tr>
</tbody>
</table>
<ul>
<li><code>return_plot = &#39;YES&#39;/NULL</code> allows you to decide whether
to plot the distances (‘YES’) or not (NULL)</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="sc">$</span> pca <span class="ot">=</span> <span class="fu">read_eigen</span>(<span class="at">pca_input =</span> <span class="st">&#39;data/TOY.pca.evec&#39;</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="sc">$</span> Source_D <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&#39;data/Sources_Distances&#39;</span>, <span class="at">header =</span>T)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="sc">$</span> <span class="fu">pcs_distances</span>(<span class="at">pca_input =</span> pca, <span class="at">output_name =</span> <span class="st">&#39;my/res/dir/output_file&#39;</span>, <span class="at">sources_file =</span> Source_D, <span class="at">return_plot =</span> <span class="st">&#39;YES&#39;</span>)</span></code></pre></div>
</div>
<div id="resampling" class="section level1">
<h1>Resampling</h1>
<p>To produce Standard Errors (SE), you can rely on resampling
techniques. Let’s say we produce 20 PCAs to estimate the SE, each of
them on a dataset with a different SNPs subset: we thus end up with 20
PCAs, all with slighly different values of PC coordinates.<br />
The function <code>read.resampling()</code> reads all 20 PCAs and
perform PANE on each of them. It will return a list containing the
results for each resampling (20 PANE results). Given that
<code>read.resampling()</code> will read and perform PANE on each
resampled set, the function uses several arguments:</p>
<p>I)<code>path_tofiles</code>: path to the directory where all the 20
PCAs are stored</p>
<p>II)<code>file_pattern</code>, a string containing the name shared by
all 20 PCAs, the function will use the string as a pattern to find all
files with said string. If this parameter seems confusing, consider that
we modelled the function considering that in a resampling setting one
could name the PCAs as follows: ‘PCA_Resampled_1’,
‘PCA_Resampled_2’,…,‘PCA_Resampled_20’. A file_patters as
‘PCA_Resampled_’ will find all files from ‘PCA_Resampled_1’ to
‘PCA_Resampled_20’.</p>
<p>III)<code>as_file</code>, this is the AS_file we used in
<code>pane()</code>. It is a two-columns file with the population list
on the first column, and the ‘A/S’ information on the second column. The
‘A/S’ information stands for Admixed (A) or Source (S), for each
population/group we will indicate whether PANE should consider it as a
Source (S) or as an admixed target (A), the file looks like this:</p>
<table>
<thead>
<tr class="header">
<th>POP A/S</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Source1</td>
<td>S</td>
</tr>
<tr class="even">
<td>Source2</td>
<td>S</td>
</tr>
<tr class="odd">
<td>Admix1</td>
<td>A</td>
</tr>
</tbody>
</table>
<ol start="4" style="list-style-type: upper-roman">
<li><code>eigentype</code>, this is an optional argument. If present,
PCA will be read through read_eigen() function, if absent PCA will be
loaded via <code>read_flash()</code>.</li>
</ol>
<p>All in all, the function can be ran as such:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="sc">$</span> AS_file <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&#39;data/Example_AS&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="sc">$</span> PANE_resampling <span class="ot">=</span> <span class="fu">read.resampling</span>(<span class="at">path_tofiles =</span> <span class="st">&#39;data/&#39;</span>, <span class="at">file_pattern =</span> <span class="st">&#39;TOY_Jack&#39;</span>, <span class="at">as_file =</span> AS_file, eigentype)</span></code></pre></div>
<p>Once you ran PANE on all 20 of your resampled PCAs, to estimate the
SE you can use the <code>se.resampling()</code> function. To do so, you
will need: * your main PANE results, obtained with <code>pane()</code>
on the main PCA (the PCA with all the available SNPs), * the PANE
results obtained from the 20 resampled PCAs (with the function
<code>read.resampling()</code> you can see above).<br />
In the example below, you can see that we first loaded the AS_file
through <code>read.table()</code>, the PANE main results are obtained
with <code>pane()</code>, the PANE resampling results are obtained with
<code>read.resampling()</code>. Importantly, to ran
<code>se.resampling()</code>, you will also need a numeric vector
containing the number of SNPs per each chromosome (ie. chromovec =
rep(1000,times = 4), 1000 are the number of SNPs used per each 4 PCA ran
for this tutorial).</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a><span class="sc">$</span> main_pca <span class="ot">=</span> <span class="fu">read_eigen</span>(<span class="at">pca_input =</span> <span class="st">&#39;data/TOY.pca.evec&#39;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="sc">$</span> AS_file <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&#39;data/Example_AS&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="sc">$</span> PANE_main_result <span class="ot">&lt;-</span> <span class="fu">pane</span>(<span class="at">pca_input =</span> main_pca, <span class="at">as_file =</span> AS_file)</span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="sc">$</span> PANE_resampling <span class="ot">=</span> <span class="fu">read.resampling</span>(<span class="at">path_tofiles =</span> <span class="st">&#39;data/&#39;</span>, <span class="at">file_pattern =</span> <span class="st">&#39;TOY_Jack&#39;</span>, <span class="at">as_file =</span> AS_file, eigentype)</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a><span class="sc">$</span> iterations <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">1000</span>,<span class="at">times =</span> <span class="dv">4</span>)</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="sc">$</span> PANE_SE <span class="ot">=</span> <span class="fu">se.resampling</span>(PANE_main_result, PANE_resampling, <span class="at">chromovec =</span> iterations)</span></code></pre></div>
</div>
<div id="evaluate-models" class="section level1">
<h1>Evaluate Models</h1>
<p>The function <code>evaluate_models()</code> tests all possible models
and outputs a list of unfit models, which are indicated by negative
performance scores. Additionally, it will return the best fit model,
allowing you to easily identify the most effective solution for your
analysis. To harness <code>evaluate_models()</code> effectively, it’s
essential to use a combination of several possible sources and targets,
ensuring a comprehensive evaluation of the models.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="sc">$</span> main_pca <span class="ot">=</span> <span class="fu">read_eigen</span>(<span class="at">pca_input =</span> <span class="st">&#39;data/TOY.pca.evec&#39;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="sc">$</span> AS_file <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&#39;data/Example_AS&#39;</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="sc">$</span> PANE_models <span class="ot">&lt;-</span> <span class="fu">evaluate_models</span>(<span class="at">pca_input =</span> main_pca, <span class="at">as_file =</span> AS_file) <span class="co">#OR</span></span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="sc">$</span> PANE_models <span class="ot">&lt;-</span> <span class="fu">evaluate_models</span>(<span class="at">pca_input =</span> main_pca, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;AFR1&quot;</span>,<span class="st">&quot;AFR2&quot;</span>,<span class="st">&quot;EUR1&quot;</span>,<span class="st">&quot;EUR2&quot;</span>,<span class="st">&quot;ASN1&quot;</span>,<span class="st">&quot;ASN2&quot;</span>), <span class="at">admixed =</span> <span class="fu">c</span>(<span class="st">&quot;70GST1.30GST2&quot;</span>)) </span></code></pre></div>
<p>The function output will list all iterations tested along with the
residuals. while negative values indicate an unfit model, also residuals
can be insightful on the model. High residuals indicate that the tested
model is not optimal, between multiple model with different residuals we
suggest to select the ones with the smallest residuals values.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
